src/main/java/com/mathworks/ci/RunMatlabBuildBuilder.java

```diff
@@ -93,16 +94,23 @@ public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
         // Invoke MATLAB build and transfer output to standard
         // Output Console
 
+        buildResult = execMatlabCommand(workspace, launcher, listener, env, build);
 
-        buildResult = execMatlabCommand(workspace, launcher, listener, env);
+        //Add build result action
+        FilePath jsonFile = new FilePath(workspace, ".matlab/buildArtifact.json");
+        if(jsonFile.exists()){
+            jsonFile.copyTo(new FilePath(new File(build.getRootDir().getAbsolutePath()+"/buildArtifact.json")));
+            jsonFile.delete();
```

Comment:
Still curious about how this works. If the workspace directory is cleared between builds or something will the dashboard go away?

Reply:
This file will be copied just after run build is complete and then run build itself invoke action followed by copy. I see less or no chances that any one could delete this file programmatically or otherwise. since this file comes from MATLAB code it does not have direct access to root folder. through MATLAB i could copy the file either in workspace somewhere or in temp directory. With temp directory deletion of file manually is possible by some othr process.